insonmnia/node/tokens.go

```diff
@@ -38,6 +39,30 @@ func (t *tokenAPI) Balance(ctx context.Context, _ *sonm.Empty) (*sonm.BalanceRep
 	}, nil
 }
 
+func (t *tokenAPI) Deposit(ctx context.Context, amount *sonm.BigInt) (*sonm.Empty, error) {
+	if err := t.remotes.eth.MasterchainToken().IncreaseApproval(ctx, t.remotes.key, blockchain.GatekeeperMasterchainAddr(), amount.Unwrap()); err != nil {
```

Comment:
either you have a bug here or docs of the increase approval are misleading. they say "approve should be called when current allowance = 0." (if it's true the API is still shit)

Reply:
The docs mean that an `.Approve()` method itself must be called with the current allowance set to zero, increase\decrease works well with any current approval value. 
I totally agree that method's docs describe the behavior not clearly, so I can fix it with an upcoming PR.